### ISSUE TITLE
Global native resolution state vector creation issue

### ISSUE THREAD

**Issue by eastjames** [geoschem/integrated_methane_inversion#181]
### Name and Institution (Required)

Name: James East
Institution: Harvard ACMG

### Description of your issue or question

I'm using the `feature/global_inversion` branch. When creating a native resolution state vector on the 2x2.5° grid, IMI creates a state vector element for all grid cells north of 60°S, totaling 10,800 state vector elements. A map looks like this:

```python
import xarray as xr
import matplotlib.pyplot as plt
import pycno

with xr.open_dataset('StateVector_orig.nc') as ds:
    sv_orig = ds['StateVector']

cno = pycno.cno(xlim=(-180,180),ylim=(-90,90))

sv_orig.plot()
cno.draw()
```
<img width="584" alt="Screenshot 2023-12-28 at 11 26 39 AM" src="https://github.com/geoschem/integrated_methane_inversion/assets/47705516/8afa9c45-b366-4fb3-835b-b99072dbaa45">

This happens in `src/utilities/make_state_vector_file.py`. The latitude of the HEMCO file used to filter for offshore O&G emissions gets manually adjusted, probably for regional applications. Also, the check for a land threshold always evaluates to `false` probably not intentionally.

Adding a check for mismatched longitudes at line 88-90 and changing the land_threshold check at line 139 seems to work:
```python
    # Require hemco diags on same global grid as land cover map
    if np.abs(lc.lon.values - hd.lon.values).max() != 0: #JDE add a check
        hd["lon"] = hd["lon"] - 0.03125  # initially offset by 0.03125 degrees
```

```python
    #if land_threshold: #old
    if land_threshold > 0: #JDE change
```

New state vector file:
<img width="577" alt="Screenshot 2023-12-28 at 11 26 50 AM" src="https://github.com/geoschem/integrated_methane_inversion/assets/47705516/a30afb0d-e9bc-44c6-be7e-85a4b93691f4">

**Comment by nicholasbalasus** [comment]
Hey James, this was due to an issue in HEMCO that we [fixed](https://github.com/geoschem/HEMCO/issues/228). See [this](https://github.com/geoschem/integrated_methane_inversion/issues/141) IMI issue. The solution is to update the HEMCO standalone outputs used for this check using the updated version of HEMCO, which we should probably do before IMI 2.0 anyway given the emissions that have changed recently.

**Comment by eastjames** [comment]
Thanks, Nick. I think this is slightly different -- the latitude adjustment was being applied regardless of what grid resolution is being used, so in global applications it was incorrectly shifting the latitude. I added a check so that it won't apply the adjustment if there is no misalignment. It sounds like updating the HEMCO standalone outputs and getting ride of the latitude adjustment will take care of it, but I needed to do this for now.
